Modules/Office365DSC/DSCResources/MSFT_SPOHomeSite/MSFT_SPOHomeSite.psm1

```diff
@@ -0,0 +1,197 @@
+        [String]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.String]
+        $Url,
+
+        [Parameter()]
+        [ValidateSet("Present", "Absent")]
+        [System.String]
+        $Ensure = "Present",
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Getting configuration for hub site collection $Url"
+
+    Test-MSCloudLogin -O365Credential $GlobalAdminAccount `
+        -Platform PnP
+
+    $nullReturn = @{
+        $IsSingleInstance  = $IsSingleInstance
+        Ensure             = "Absent"
+        GlobalAdminAccount = $GlobalAdminAccount
+    }
+
+    try
+    {
+        Write-Verbose -Message "Getting current home site collection settings"
+        $homeSiteUrl = Get-SPOHomeSite
+        if ($null -eq $homeSiteUrl)
+        {
+            Write-Verbose -Message "There is no Home Site Collection set."
+            return $nullReturn
+        }
+        else
+        {
+            $result = @{
+                $IsSingleInstance  = $IsSingleInstance
+                $Url               = $homeSiteUrl
+                Ensure             = "Present"
+                GlobalAdminAccount = $GlobalAdminAccount
+            }
+            return $result
+        }
+    }
+    catch
+    {
+        Write-Verbose -Message "There was an error in the SPOHomeSite resource."
+    }
+    return $nullReturn
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateSet("Yes")]
+        [String]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.String]
+        $Url,
+
+        [Parameter()]
+        [ValidateSet("Present", "Absent")]
+        [System.String]
+        $Ensure = "Present",
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Setting configuration for home site '$Url'"
+
+    Test-MSCloudLogin -O365Credential $GlobalAdminAccount `
+        -Platform SharePointOnline
+
+    Test-MSCloudLogin -O365Credential $GlobalAdminAccount `
+        -Platform PnP
+
+    $currentValues = Get-TargetResource @PSBoundParameters
+
+    if ($Ensure -eq "Present" -and $currentValues.Ensure -eq "Absent" `
+            -or $Ensure -eq "Present" -and $currentValues.Ensure -eq "Present")
+    {
+        try
+        {
+            Write-Verbose -Message "Setting home site collection $Url"
+            Get-SPOSite $Url
+        }
+        catch
+        {
+            $Message = "The specified Site Collection $($Url) for SPOHomeSite doesn't exist."
+            New-Office365DSCLogEntry -Error $_ -Message $Message
+            throw $Message
+        }
+
+        Write-Verbose -Message "Configuring site collection as Home Site"
+        Set-PnPHomeSite -Url $Url
+    }
+    else
+    {
+        # Remove home site
```

Comment:
This should only be called if $currentValues.Ensure -eq 'Present' otherwise it's already absent and we shouldn't try to remove it